source/includes/read/project.c

```diff
@@ -0,0 +1,79 @@
+#include <bson/bson.h>
+#include <mongoc/mongoc.h>
+#include <stdio.h>
+
+int
+main (int argc, char *argv[])
+{
+    mongoc_client_t *client;
+    mongoc_collection_t *collection;
+    bson_error_t error;
+
+    mongoc_init ();
+
+    client =
+        mongoc_client_new ("<connection string URI>");
+    collection = mongoc_client_get_collection (client, "sample_restaurants", "restaurants");
+
+    {
+        // start-project
+        char *str;
+        const bson_t *doc;
+
+        bson_t *filter = BCON_NEW ("name", BCON_UTF8 ("Emerald Pub"));
+        bson_t *opts = BCON_NEW ("projection", "{",
+                            "name", BCON_BOOL (true),
+                            "cuisine", BCON_BOOL (true),
+                            "borough", BCON_BOOL (true),
+                            "}");
+
+        mongoc_cursor_t *results = 
+            mongoc_collection_find_with_opts (collection, filter, opts, NULL);
+
+        while (mongoc_cursor_next (results, &doc)) {
+            str = bson_as_canonical_extended_json (doc, NULL);
```

Comment:
Consider moving the declaration of `str` so the declaration is close to the use:

```c
char* str = bson_as_canonical_extended_json (doc, NULL);
```

Same suggestion applies below.